pennylane/_qubit_device.py

```diff
@@ -312,19 +312,23 @@ def sample_basis_states(self, number_of_states, state_probability):
         return np.random.choice(basis_states, self.shots, p=state_probability)
 
     @staticmethod
-    def states_to_binary(samples, num_wires):
+    def states_to_binary(samples, num_wires, dtype=None):
         """Convert basis states from base 10 to binary representation.
 
         This is an auxiliary method to the generate_samples method.
 
         Args:
             samples (List[int]): samples of basis states in base 10 representation
-            number_of_states (int): the number of basis states to sample from
+            num_wires (int): the number of qubits
+            dtype (type): type of the internal array to be used. Can be
+                important to specify for large systems for memory allocation
+                purposes.
 
         Returns:
             List[int]: basis states in binary representation
         """
-        powers_of_two = 1 << np.arange(num_wires)
+        wires_array = np.arange(num_wires) if dtype is None else np.arange(num_wires, dtype=dtype)
+        powers_of_two = 1 << wires_array
```

Comment:
With the default `dtype` specified, you can then simply do
```suggestion
        powers_of_two = 1 << np.arange(num_wires, dtype=dtype)
```